### PR TITLE
Add firebase crashlytics

### PR DESCRIPTION
**Pull request by mysticetii**
Added firebase crashlytics

### CODE REVIEW

File: android/app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -1,9 +1,13 @@
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
     package="uk.geovation.flutter_sandbox">
+    <uses-permission android:name="android.permission.INTERNET"/>
+    <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" />
```
Comment (sebastianovide) [inline android/app/src/main/AndroidManifest.xml:4]: is that required for crashlitics ?

File: lib/firebase_crashlytics/firebase_crashlytics_page.dart [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+class FirebaseCrashlyticsPage extends StatelessWidget {
+  static const id = 'firebase_crashlytics_page';
+
+  @override
+  Widget build(BuildContext context) {
+    FirebaseCrashlytics fci = FirebaseCrashlytics.instance;
+    String message =
+        'Press the button to simulate a crash. Restart the app for crashlytics to log the crash';
+
+    Function onPressCrash = () async {
```
Comment (sebastianovide) [inline lib/firebase_crashlytics/firebase_crashlytics_page.dart:14]: nice... good idea !

**Review by sebastianovide — COMMENTED** [review]
any way that you can add the user id to crashlytic when the user logs in/out ?

File: android/app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (mysticetii) [reply android/app/src/main/AndroidManifest.xml:4]: It actually handles an error message from mapbox_gl, when I was looking for platformexception bug on android 11.

**Comment by mysticetii** [comment]
> any way that you can add the user id to crashlytic when the user logs in/out ?

I have added userid for crashlytics and if user is logged out a default value for id is showns.

File: lib/firebase_crashlytics/firebase_crashlytics_page.dart [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+      FirebaseAuth _auth = FirebaseAuth.instance;
+      await fci.setCrashlyticsCollectionEnabled(true);
+      if (fci.isCrashlyticsCollectionEnabled) {
+        if (_auth.currentUser != null) {
+          FirebaseCrashlytics.instance.setUserIdentifier(_auth.currentUser.uid);
+        } else {
+          FirebaseCrashlytics.instance.setUserIdentifier("logged_out_user");
```
Comment (sebastianovide) [inline lib/firebase_crashlytics/firebase_crashlytics_page.dart:22]: change it to blank string

Comment (mysticetii) [reply lib/firebase_crashlytics/firebase_crashlytics_page.dart]: logged_out_user string has been modified to empty string in the next commit

**Pull request merged**